provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/UpgradeGuidanceRiskModel.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * provider
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.provider.blackduck.processor.message.service;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.alert.common.rest.model.AlertSerializableModel;
+import com.synopsys.integration.blackduck.api.generated.component.ComponentVersionUpgradeGuidanceLongTermVulnerabilityRiskView;
+import com.synopsys.integration.blackduck.api.generated.component.ComponentVersionUpgradeGuidanceShortTermVulnerabilityRiskView;
+
+public class UpgradeGuidanceRiskModel extends AlertSerializableModel {
+    private final Integer critical;
+    private final Integer high;
+    private final Integer medium;
+    private final Integer low;
```

Comment:
Why did we decide to use Integer? Can we get away with the use of int and defaulting everything to 0? Do we actually benefit from a null value being returned as opposed to a 0?

Reply:
This code was mostly copied/adapted from some old code handling this. I was trying to keep feature parity. These fields can be null in the response we get from BlackDuck, so I'm guessing we were trying to give a similar experience.